Client/Frontend/Browser/PageZoom/PageZoomView.swift

```diff
@@ -0,0 +1,208 @@
+    HStack(spacing: 0.0) {
+      Button(action: {
+        onDecrement()
+      }) {
+        Image(systemName: "minus")
+          .font(.system(.footnote).weight(.medium))
+          .foregroundColor(color)
+          .frame(width: 44.0)
+          .padding(.vertical, 8.0)
+          .contentShape(Rectangle())
+      }
+      .fixedSize()
+      .disabled(value == minValue)
+      
+      Divider()
+        .padding(.vertical, 8.0)
+      
+      Text(PageZoomView.percentFormatter.string(from: .init(value: 0.999)) ?? "999%")
+        .font(.system(.footnote).weight(.medium))
+        .fixedSize()
+        .hidden()
+        .overlay(resetZoomButton)
+        .padding(.horizontal)
+        .padding(.vertical, 8.0)
+      
+      Divider()
+        .padding(.vertical, 8.0)
+      
+      Button(action: {
+        onIncrement()
+      }) {
+        Image(systemName: "plus")
+          .font(.system(.footnote).weight(.medium))
+          .foregroundColor(color)
+          .frame(width: 44.0)
+          .padding(.vertical, 8.0)
+          .contentShape(Rectangle())
+      }
+      .fixedSize()
+      .disabled(value == maxValue)
+    }
+    .fixedSize(horizontal: false, vertical: true)
+    .background(Color(UIColor.secondaryBraveBackground))
+    .clipShape(RoundedRectangle(cornerRadius: 10.0, style: .continuous))
+  }
+  
+  private var resetZoomButton: some View {
+    Button(action: {
+      onReset()
+    }) {
+      Text(PageZoomView.percentFormatter.string(from: .init(value: value)) ?? "\(Int(value * 100.0))%")
+        .font(.system(.footnote).weight(.medium))
+        .foregroundColor(color)
+        .padding()
+        .contentShape(Rectangle())
+    }
+    .fixedSize()
+    .disabled(value == 1.0)
+  }
+}
+
+struct PageZoomView: View {
+  @Environment(\.managedObjectContext) private var context
+  
+  private weak var webView: WKWebView?
+  @State private var minValue = 0.5
+  @State private var maxValue = 3.0
+  @State private var currentValue: Double
+  
+  public static let percentFormatter = NumberFormatter().then {
+    $0.numberStyle = .percent
+    $0.minimumIntegerDigits = 1
+    $0.maximumIntegerDigits = 3
+    $0.maximumFractionDigits = 0
+    $0.minimumFractionDigits = 0
+  }
+  
+  public static let propertyName = "viewScale"
+  
+  public static let steps = [0.5, 0.75, 0.85,
+                             1.0, 1.15, 1.25,
+                             1.50, 1.75, 2.00,
+                             2.50, 3.0]
+  
+  var dismiss: (() -> Void)?
+  
+  init(webView: WKWebView?) {
+    self.webView = webView
+    
+    if let url = webView?.url,
+       let domain = Domain.getPersistedDomain(for: url) {
+      
+      _currentValue = State(initialValue: domain.zoom_level?.doubleValue ?? Preferences.General.defaultPageZoomLevel.value)
+    } else {
+      _currentValue = State(initialValue: webView?.value(forKey: PageZoomView.propertyName) as? Double ?? Preferences.General.defaultPageZoomLevel.value)
+    }
```

Comment:
Maybe leave a comment here mentioning that this is OK because you are only presenting `PageZoomView` via `UIHostingController` so there's no chance of re-init.